usersession/agent/rest_api.go

```diff
@@ -461,10 +462,19 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 		notification.WithUrgency(notification.LowUrgency),
 	}
 
+	// But if we have a desktop file of the busy application, use that apps's icon.
+	if finishRefresh.AppDesktopEntry != "" {
+		parser := goconfigparser.New()
+		desktopFilePath := filepath.Join(dirs.SnapDesktopFilesDir, finishRefresh.AppDesktopEntry+".desktop")
```

Comment:
Given what I wrote in the earlier comment, this only works by accident. The details on how desktop file names are constructed are here https://github.com/snapcore/snapd/blob/b04ff60f9b12b792259c066a243c855e8f879173/snap/info.go#L839 and https://github.com/snapcore/snapd/blob/b04ff60f9b12b792259c066a243c855e8f879173/wrappers/desktop.go#L269

One option would be to glob for `<instance-name>_*.desktop` to locate the files, but then there can be more than one file so you'll likely need some heuristic to pick the one which specifies the 'main' icon you may want to use in the notification. Maybe it'd be enough to call `snap.SplitInstnaceName()` then use the snap name and pick the desktop file named `<instance-name>_<snap-name>.desktop`. For instance, for `snap-store`, this would be `snap-store_snap-store.desktop`, or assuming a paralle installed `snap-store_foo`, the file would be `snap-store+foo_snap-store.desktop`.

Reply:
Ok... something like this...?

Reply:
you can do this:
```
si, err := snap.ReadCurrentSnapInfo(finishRefresh.InstanceName)
..
// trivial heuristic, if the app is named like a snap then
// it's considered to be the main user facing app and hopefully carries
// a nice icon
mainApp, ok := si.Apps[si.SnapName()]
desktopFile := ""
if ok && !mainApp.IsService() {
     // got the main app, grab its desktop file
    desktopFile = mainApp.DesktopFile()
} else {
    // pick some other app or nothing?
    .. // set desktopFile as needed
}
if desktopFile != "" {
    // load it, find the Icon

}
```

Reply:
Done.